cloud/openstack/os_ironic_inspect.py

```diff
@@ -0,0 +1,167 @@
+            type: string
+            sample: "10"
+        cpus:
+            description: Count of cpu cores defined in the updated node properties.
+            type: string
+            sample: "1"
+'''
+
+EXAMPLES = '''
+# Invoke node inspection
+- os_ironic_inspect:
+    name: "testnode1"
+'''
+
+
+def _choose_id_value(module):
+    if module.params['uuid']:
+        return module.params['uuid']
+    if module.params['name']:
+        return module.params['name']
+    return None
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        auth_type=dict(required=False),
+        uuid=dict(required=False),
+        name=dict(required=False),
+        mac=dict(required=False),
+        ironic_url=dict(required=False),
+        timeout=dict(default=1200, type='int', required=False),
+    )
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(argument_spec, **module_kwargs)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+    if StrictVersion(shade.__version__) < StrictVersion('1.0.0'):
+        module.fail_json(msg="To utilize this module, the installed version of"
+                             "the shade library MUST be >=1.0.0")
+
+    if (module.params['auth_type'] in [None, 'None'] and
+            module.params['ironic_url'] is None):
+        module.fail_json(msg="Authentication appears to be disabled, "
+                             "Please define an ironic_url parameter")
+
+    if (module.params['ironic_url'] and
+            module.params['auth_type'] in [None, 'None']):
+        module.params['auth'] = dict(
+            endpoint=module.params['ironic_url']
+        )
+
+    try:
+        cloud = shade.operator_cloud(**module.params)
+
+        if module.params['name'] or module.params['uuid']:
+            server = cloud.get_machine(_choose_id_value(module))
+        elif module.params['mac']:
+            server = cloud.get_machine_by_mac(module.params['mac'])
+        else:
+            module.fail_json(msg="The worlds did not align, "
+                                 "the host was not found as "
+                                 "no name, uuid, or mac was "
+                                 "defined.")
+        if server:
+            cloud.inspect_machine(server['uuid'], module.params['wait'])
+            # TODO(TheJulia): diff properties, ?and ports? and determine
+            # if a change occured.  In theory, the node is always changed
+            # if introspection is able to update the record.
+            module.exit_json(changed=True,
+                             ansible_facts=server['properties'])
+
+        else:
+            module.fail_json(msg="node not found.")
+
+    except shade.OpenStackCloudException as e:
+        module.fail_json(msg=str(e))
+
+
+# this is magic, see lib/ansible/module_common.py
+from ansible.module_utils.basic import *
+from ansible.module_utils.openstack import *
+main()
```

Comment:
Ansible folks are going to insist on adding this around the call to main:

if __name__ == "__main__":